Quaver.Shared/Screens/Edit/Input/EditorInputConfig.cs

```diff
@@ -0,0 +1,184 @@
+                Keybinds[action].Add(keybind);
+        }
+
+        public bool RemoveKeybindFromAction(KeybindActions action, Keybind keybind)
+        {
+            if (Keybinds.ContainsKey(action))
+                return Keybinds[action].Remove(keybind);
+            return false;
+        }
+
+        public void SaveToConfig()
+        {
+            File.WriteAllText(ConfigPath, Serialize());
+            Logger.Debug("Saved editor key config to file", LogType.Runtime);
+        }
+
+        public void OpenConfigFile()
+        {
+            try
+            {
+                Utils.NativeUtils.OpenNatively(ConfigPath);
+            }
+            catch (Exception e)
+            {
+                Logger.Debug(e.ToString(), LogType.Runtime);
+            }
+        }
+
+        public void FillMissingKeys(bool fillWithDefaultBinds)
+        {
+            var count = 0;
+
+            foreach (var (action, defaultBind) in DefaultKeybinds)
+            {
+                if (!Keybinds.ContainsKey(action))
+                {
+                    var bind = fillWithDefaultBinds ? defaultBind : new KeybindList();
+                    Keybinds.Add(action, bind);
+                    count++;
+                }
+            }
+
+            if (count > 0)
+            {
+                SaveToConfig();
+                Logger.Debug($"Filled {count} missing action keybinds in key config file", LogType.Runtime);
+            }
+        }
+
+        public void ResetConfigFile()
+        {
+            ReverseScrollSeekDirection = true;
+            Keybinds = DefaultKeybinds;
+            PluginKeybinds = new Dictionary<string, KeybindList>();
+            SaveToConfig();
+            Logger.Debug("Reset editor keybind config file", LogType.Runtime);
+        }
+
+        public Dictionary<Keybind, HashSet<KeybindActions>> ReverseDictionary()
+        {
+            var dict = new Dictionary<Keybind, HashSet<KeybindActions>>();
+
+            foreach (var (action, keybinds) in Keybinds)
+            {
+                foreach (var keybind in keybinds.MatchingKeybinds())
+                {
+                    if (dict.ContainsKey(keybind))
+                        dict[keybind].Add(action);
+                    else
+                        dict[keybind] = new HashSet<KeybindActions>() {action};
+                }
+            }
+
+            return dict;
+        }
+
+        private static EditorInputConfig Deserialize(StreamReader file)
+        {
+            var ds = new DeserializerBuilder()
+                .WithTypeConverter(new KeybindYamlTypeConverter())
+                .IgnoreUnmatchedProperties()
+                .Build();
+
+            EditorInputConfig config;
+
+            try
+            {
+                config = ds.Deserialize<EditorInputConfig>(file);
+            }
+            catch (YamlException e)
+            {
+                throw new Exception($"Failed to parse configuration in line {e.Start.Line}");
```

Comment:
Kinda defeats the purpose of the try catch if it's going to throw another exception? Maybe just return a fresh instance of the config and log the error.